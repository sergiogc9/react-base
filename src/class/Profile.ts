export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
