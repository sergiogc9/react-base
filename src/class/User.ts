export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
