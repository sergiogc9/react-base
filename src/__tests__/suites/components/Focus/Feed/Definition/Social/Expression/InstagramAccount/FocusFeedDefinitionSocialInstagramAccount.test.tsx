import React from "react";
import { shallow } from "enzyme";

import FocusFeedDefinitionSocialInstagramAccount from "@src/components/Focus/Feed/Definition/Social/Expression/InstagramAccount/FocusFeedDefinitionSocialInstagramAccount";
import { SocialInstagramAccount } from "@src/class/Feed";

const instagramAccount: SocialInstagramAccount = {
	id: "id1",
	name: "Girona FC",
	profile_picture_url: "fakeurl.com",
	screen_name: "gironafc",
	valid: true
};

describe("Definition social instagram account", () => {
	let wrapper: any;
	let onSelected = jest.fn();

	const getComponent = () => (
		<FocusFeedDefinitionSocialInstagramAccount
			account={instagramAccount}
			removable={false}
			onSelected={onSelected}
		/>
	);

	beforeEach(() => {
		onSelected = jest.fn();
		wrapper = shallow(getComponent());
	});

	it("Component default snaphsot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snaphsot with removable props", () => {
		wrapper.setProps({ removable: true });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snaphsot with not valid account", () => {
		wrapper.setProps({ account: { ...instagramAccount, valid: false } });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("On remove expression", () => {
		expect(onSelected).toHaveBeenCalledTimes(0);
		wrapper.find(`#focusFeedDefinitionSocialInstagramAccount${instagramAccount.id}`).at(0).simulate('click');
		expect(onSelected).toHaveBeenCalledTimes(1);
	});
});
