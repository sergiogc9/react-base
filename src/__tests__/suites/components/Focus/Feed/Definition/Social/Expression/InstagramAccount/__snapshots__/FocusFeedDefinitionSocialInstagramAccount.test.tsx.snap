// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Definition social instagram account Component default snaphsot 1`] = `"<div id=\\"focusFeedDefinitionSocialInstagramAccountid1\\" class=\\"instagram-account \\"><img src=\\"fakeurl.com\\" alt=\\"\\"/><span>Girona FC</span><span>(@gironafc)</span></div>"`;

exports[`Definition social instagram account Component snaphsot with not valid account 1`] = `"<div id=\\"focusFeedDefinitionSocialInstagramAccountid1\\" class=\\"instagram-account not_valid\\" title=\\"feed.form.instagram.expired_account\\"><img src=\\"fakeurl.com\\" alt=\\"\\"/><span>Girona FC</span><span>(@gironafc)</span><span class=\\"no-line-through\\"> [expired]</span></div>"`;

exports[`Definition social instagram account Component snaphsot with removable props 1`] = `"<div id=\\"focusFeedDefinitionSocialInstagramAccountid1\\" class=\\"instagram-account \\"><img src=\\"fakeurl.com\\" alt=\\"\\"/><span>Girona FC</span><span>(@gironafc)</span><i class=\\"md-icon material-icons\\">clear</i></div>"`;
